languages/ruby/lib/oso/polar/query.rb

```diff
@@ -71,6 +72,9 @@ def application_error(message)
       # @raise [Error] if the FFI call raises one.
       def handle_call(attribute, call_id:, instance:, args:, kwargs:) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
         instance = host.to_ruby(instance)
+        rel = get_relationship(instance.class, attribute)
+        return handle_relationship(call_id, instance, rel) unless rel.nil?
```

Comment:
I think I'm confused -- why does this need to be handled here? Shouldn't the VM be the one handling relationships?

Does the host library handle relationships on bound variables while the VM handles relationships on unbound variables?

```
allow(user, "read", document) if
  folder in user.folders and
#               ^ here, user is bound
  document in folder.documents;
#                   ^ here, folder is not bound
```

Reply:
no, it's not about bound vs. unbound. if we end up storing type information in the VM then we could do some of the work there. but here a relationship could be handled by an ORM or something, so it has to be handled in the host.

Reply:
Ohh I think I maybe understand this now. Question: if we were to get rid of this, data filtering would still work, right? It just would potentially eagerly load one "layer" of relationships?

```
allow(user, "read", document) if
  folder in user.folders and
  document in folder.documents;
```
^ For example, here `user.folders` would end up bound to an array of folders, instead of being bound to a variable?

Reply:
no, it would break. a relationship is kind of like a "virtual" field on a type, it doesn't necessarily exist on the external instance, but the host library handles it so it looks as if it did.